app/controllers/shopify_app/callback_controller.rb

```diff
@@ -69,7 +69,7 @@ def update_rails_cookie(api_session, cookie)
     end
 
     def redirect_to_app
-      if ShopifyAPI::Context.embedded?
+      if ShopifyAPI::Context.embedded? && !fully_formed_url?(session[:return_to])
         return_to = "#{decoded_host}#{session.delete(:return_to)}"
         return_to = ShopifyApp.configuration.root_url if deduced_phishing_attack?
```

Comment:
We'll still need this `deduced_phishing_attack?` check here as well. The `return_to` param is used with phishing attacks to make links seem legit because they linked back from shopify.